src/instrumentation.ts

```diff
@@ -425,14 +446,15 @@ export class BullMQInstrumentation extends InstrumentationBase {
         const currentContext = context.active();
         const producerContext = propagation.extract(currentContext, job.opts);
 
-        const spanName = `${job.queueName}.${job.name} Worker.${workerName} #${job.attemptsMade}`;
+        const spanName = `${job.queueName} ${operationType}`;
```

Comment:
Just some thoughts about extractors and the tracing project:

Ah that's the OpenTelemetry standard for span naming here? I hoped that it was a thing that the other background worker library did I tested and I could change later.

To replicate the behavior we have now of setting the action name to something like `WorkerClassName#perform`, do we read from the `SemanticAttributes.MESSAGING_CONSUMER_ID` or `BullMQAttributes.JOB_NAME` attribute?

Reply:
> To replicate the behavior we have now of setting the action name to something like WorkerClassName#perform

We can't really do that here, we don't have that information. (On Ruby, I suppose you could extract that information from code location spans, if they are emitted by your instrumentation)

Furthermore, in BullMQ, workers don't even have names! You declare a worker like:

```javascript
Worker.run("queueName", (job) => {
  ...
});
```

That is, the closest thing it has to a name is that of the queue it listens to. (This is why I've removed a couple of "worker name" and "queue name" attributes -- they were all showing the same value, the name of the queue)

In the not-yet-published version 1.26.0 of the OpenTelemetry Semantic Conventions, you will be allowed to use the `messaging.operation.name` (that is, `"Worker.run"`) instead of the `messaging.operation.type` (that is, `"process"`) as part of the span name. This will be an improvement. (In the meantime, we'll fix it in post for AppSignal users)

Reply:
What will it do in the extractor then to set the action name? Use the queue name?

Reply:
Yeah, the action name for the `Worker.run` span will use the queue name. (The job name, unfortunately, is potentially high-cardinality)

I was thinking something like:
- Category: `${messaging.operation.type}.bullmq`
  - e.g.: `publish.bullmq`, `process.bullmq`
- Name: `${messaging.operation.name} (${messaging.destination})`
  - e.g. `Queue.addBulk (queueName)`, `Worker.run (queueName)`)

Reply:
How high-cardinality will the job name be? We're fine with "Worker#function", "Queue function", but not "JobID123". Is it just whatever people give to "Worker.run" and that can be anything?

Reply:
It's whatever people give when enqueueing the job:

```js
const queue = new Queue("queueName");
await q.add("jobName", { arbitrary: "job data" })
```

It can be anything, and it's meant to be "human-meaningful", so it could be "send stroopwafels to customer #12463".